homeassistant/components/camera/__init__.py

```diff
@@ -715,7 +715,9 @@ async def get(self, request: web.Request, entity_id: str) -> web.StreamResponse:
         )
 
         if not authenticated:
-            raise web.HTTPUnauthorized()
+            if request[KEY_AUTHENTICATED]:
+                raise web.HTTPUnauthorized()
```

Comment:
This line will never reached since authenticated will always true if request[KEY_AUTHENTICATED] was true. See line 712